src/odemis/acq/fastem.py

```diff
@@ -962,49 +963,54 @@ def estimateTiledAcquisitionTime(stream, stage, area):
+            The x and y axes are aligned with the x and y axes of the ebeam scanner.
+        :param area: (float, float, float, float) xmin, ymin, xmax, ymax coordinates of the overview region.
+        :param live_stream: (StaticStream or None): StaticStream to be updated with each tile acquired,
+               to build up live the whole acquisition. NOT SUPPORTED YET.
 
-    def _pass_future_progress(sub_f, start, end):
-        f.set_progress(start, end)
+        :returns: (DataArray) The complete overview image.
+        """
+        fastem_conf.configure_scanner(stream.emitter, fastem_conf.OVERVIEW_MODE)
 
-    # Note, for debugging, it's possible to keep the intermediary tiles with log_path="./tile.ome.tiff"
-    sf = stitching.acquireTiledArea([stream], stage, area, overlap, registrar=REGISTER_IDENTITY,
-                                    focusing_method=FocusingMethod.NONE)
-    # Connect the progress of the underlying future to the main future
-    # FIXME removed to provide better progress update in GUI
-    #  When _tiledacq.py has proper time update implemented, add this line here again.
-    # sf.add_update_callback(_pass_future_progress)
-    das = sf.result()
+        # The stage movement precision is quite good (just a few pixels). The stage's
+        # position reading is much better, and we can assume it's below a pixel.
+        # So as long as we are sure there is some overlap, the tiles will be positioned
+        # correctly and without gap.
+        overlap = STAGE_PRECISION / stream.emitter.horizontalFoV.value
+        logging.debug("Overlap is %s%%", overlap * 100)  # normally < 1%
 
-    if len(das) != 1:
-        logging.warning("Expected 1 DataArray, but got %d: %r", len(das), das)
+        # Note, for debugging, it's possible to keep the intermediary tiles with log_path="./tile.ome.tiff"
+        self._sub_future = stitching.acquireTiledArea([stream], stage, area, overlap, registrar=REGISTER_IDENTITY,
+                                        focusing_method=FocusingMethod.NONE)
 
-    # Switch immersion mode back on, so we can focus the SEM from the TFS GUI.
-    stream.emitter.immersion.value = True
+        das = []
+        try:
+            das = self._sub_future.result()
+        finally:
+            # Switch immersion mode back on, so we can focus the SEM from the TFS GUI.
+            stream.emitter.immersion.value = True
 
-    # FIXME auto blanking not working properly, so force beam blanking after image acquisition for now.
-    stream.emitter.blanker.value = True
+            # FIXME auto blanking not working properly, so force beam blanking after image acquisition for now.
+            stream.emitter.blanker.value = True
 
-    return das[0]
+        if len(das) == 1:
+            return das[0]
+        else:
+            logging.warning("Expected 1 DataArray, but got %d: %r", len(das), das)
+            return []
```

Comment:
If there are too many DataArrays, it's nice to at least pass one, instead of discarding everything. You can do that so:
```suggestion
            return das[:1]
```